src/main/java/com/uber/cadence/samples/hello/HelloSignal.java

```diff
@@ -51,16 +53,30 @@
   /** GreetingWorkflow implementation that returns a greeting. */
   public static class GreetingWorkflowImpl implements GreetingWorkflow {
 
-    private final CompletablePromise<String> name = Workflow.newPromise();
+    private final int totalGreetings = 2;
+
+    private final CompletablePromise<Boolean> allNamesQueued = Workflow.newPromise();
+    private final List<String> names = new ArrayList<String>();
```

Comment:
I think it might be better if we use a BlockingQueue here, so that (1) we still get the blocking semantics when there is no signal sent; and (2) we don't need a pre-defined total count (which most use cases don't have as signals are mostly external) .

Reply:
adjusted this to use workflow.await which will achieve the same behavior of waiting on a specified number of signals